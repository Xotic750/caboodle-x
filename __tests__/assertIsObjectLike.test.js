import objects from './helpers/objects';
import {assertIsObjectLike} from '../index';

describe('assertIsObjectLike', () => {
  it('should be a function', () => {
    expect(assertIsObjectLike).toBeInstanceOf(Function);
  });

  it('should return original object', () => {
    const expected = objects;
    const actual = objects.map(object => assertIsObjectLike(object));

    expect(actual).toEqual(expected);
  });

  it('should throw with default message', () => {
    expect(() => {
      assertIsObjectLike(0);
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw with custom message', () => {
    expect(() => {
      assertIsObjectLike(true, 'Custom message');
    }).toThrowErrorMatchingSnapshot();
  });
});
