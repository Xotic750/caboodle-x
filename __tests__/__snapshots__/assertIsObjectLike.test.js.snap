// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertIsObject should throw with custom message 1`] = `"Custom message"`;

exports[`assertIsObject should throw with default message 1`] = `"Not an object"`;
